share/sysbuild/cmake/modules/sysbuild_snippets.cmake

```
@@ -0,0 +1,16 @@
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_get(SB_SNIPPET)
+if(NOT SB_SNIPPET AND SNIPPET)
+  set_ifndef(SB_SNIPPET ${SNIPPET})
+endif()
```

isn't
```suggestion
set_ifndef(SB_SNIPPET ${SNIPPET})
```
sufficient ?
If `SB_SNIPPET` is not set, then it will be set to SNIPPET.
If SNIPPET is undefined, the SB_SNIPPET too will stay undefined.

---

It actually gives a cmake error about the number of arguments being incorrect

---

```
CMake Error at cmake/modules/sysbuild_snippets.cmake:7 (set_ifndef):
  set_ifndef Function invoked with incorrect arguments for function named:
  set_ifndef
Call Stack (most recent call first):
  cmake/modules/sysbuild_default.cmake:18 (include)
  /tmp/aa/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:75 (include)
  /tmp/aa/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:117 (include_boilerplate)
  /tmp/aa/zephyr/share/sysbuild-package/cmake/SysbuildConfig.cmake:8 (include)
  template/CMakeLists.txt:10 (find_package)
```

---

Nevermind, `set_ifndef()` requires the value to be defined, else you'll get missing argument when `SNIPPET` is undefined.

Perhaps it would be good to allow `<value>` to be empty in `set_ifndef()` to remove the need for the construct.